packages/components/src/components/button/button.e2e-test.ts

```diff
@@ -1,66 +1,554 @@
+    });
+
+    // Soft Disabled button
+    await test.step(`attribute soft-disabled should be present on ${buttonType} button`, async () => {
+      await componentsPage.setAttributes(button, {
+        'soft-disabled': '',
+      });
+      await expect(button).toHaveAttribute('soft-disabled');
+      await componentsPage.removeAttribute(button, 'soft-disabled');
+    });
+
+    // Active button
+    await test.step(`attribute active should be present on ${buttonType} button`, async () => {
+      await componentsPage.setAttributes(button, {
+        active: '',
+      });
+      await expect(button).toHaveAttribute('active');
+      await componentsPage.removeAttribute(button, 'active');
+    });
 
-  // initial check for the button be visible on the screen:
-  await button.waitFor();
+    // Active Disabled button
+    await test.step(`attribute active and disabled should be present on ${buttonType} button`, async () => {
+      await componentsPage.setAttributes(button, {
+        disabled: '',
+        active: '',
+      });
+      await expect(button).toHaveAttribute('disabled');
+      await expect(button).toHaveAttribute('active');
+      await componentsPage.removeAttribute(button, 'disabled');
+      await componentsPage.removeAttribute(button, 'active');
+    });
 
-  /**
-   * ACCESSIBILITY
-   */
-  await test.step('accessibility', async () => {
-    await componentsPage.accessibility.checkForA11yViolations('button-default');
+    // Active Soft Disabled button
+    await test.step(`attribute active and soft-disabled should be present on ${buttonType} button`, async () => {
+      await componentsPage.setAttributes(button, {
+        'soft-disabled': '',
+        active: '',
+      });
+      await expect(button).toHaveAttribute('soft-disabled');
+      await expect(button).toHaveAttribute('active');
+      await componentsPage.removeAttribute(button, 'soft-disabled');
+      await componentsPage.removeAttribute(button, 'active');
+    });
   });
+};
 
-  /**
-   * VISUAL REGRESSION
-   */
-  await test.step('visual-regression', async () => {
-    await test.step('matches screenshot of element', async () => {
-      await componentsPage.visualRegression.takeScreenshot('mdc-button', { element: button });
+const testForButtonSizes = async (args: SetupOptions, buttonType: string) => {
+  await test.step('attribute size for icon button', async () => {
+    const { componentsPage, ...props } = args;
+    const button = await setup({
+      componentsPage,
+      ...props,
     });
+
+    if (buttonType === 'icon') {
+      for (const size of Object.values(ICON_BUTTON_SIZES)) {
+        if (size !== ICON_BUTTON_SIZES[20]) {
+          await test.step(`attribute size="${size}" should be present on ${buttonType} button`, async () => {
+            await componentsPage.setAttributes(button, { size: `${size}`, 'aria-label': 'icon-button-20' });
+            await expect(button).toHaveAttribute('size', `${size}`);
+          });
+        }
```

Comment:
[suggestion]: instead of an if condition, which checks for each iteration, we can filter the size 20 before we run the loop.
```suggestion
      const iconSizesToTest = Object.values(ICON_BUTTON_SIZES).filter(size => size !== ICON_BUTTON_SIZES[20]);
      for (const size of iconSizesToTest) {
        await test.step(`attribute size="${size}" should be present on ${buttonType} button`, async () => {
          await componentsPage.setAttributes(button, { size: `${size}`, 'aria-label': 'icon-button-20' });
          await expect(button).toHaveAttribute('size', `${size}`);
        });
      }
```
Note: this is just a suggestion, feel free to ignore.